crates/ruff/resources/test/fixtures/perflint/PERF403.py

```diff
@@ -0,0 +1,35 @@
+def foo():
+    result = {}
+    fruit = ["apple", "pear", "orange"]
+    for idx, name in enumerate(fruit):
+        result[idx] = name  # PERF403
```

Comment:
(I wonder similar to `PERF402` why there isn't the separation for dictionary, I'm assuming the corresponding rule for dictionary doesn't exists upstream as well).

Reply:
Yeah don't know why it's split into `copy()` and `comprehension` for lists but only one for dicts. My implementation was just a straight copy from https://github.com/tonybaloney/perflint/blob/main/perflint/comprehension_checker.py 

As we are already deviating with the if conditional checks, we could add a separate rule `PERF404` if that has value.